Use fsdp module to initialize precision scalar for fsdp native (#14092)
Lightning-AI/lightning: src/pytorch_lightning/CHANGELOG.md

### Added

> - Added ``FullyShardedNativeNativeMixedPrecisionPlugin`` to handle precision for ``DDPFullyShardedNativeStrategy`` ([#14092](https://github.com/Lightning-AI/lightning/pull/14092))

Reviewer: ```suggestion
- Added `FullyShardedNativeNativeMixedPrecisionPlugin` to handle precision for `DDPFullyShardedNativeStrategy` ([#14092](https://github.com/Lightning-AI/lightning/pull/14092))
```

- Fixed dtype inference during gradient norm computation ([#14051](https://github.com/Lightning-AI/lightning/pull/14051))

> - Fixed to use fsdp module to initialize precision scalar for fsdp native ([#14092](https://github.com/Lightning-AI/lightning/pull/14092))

Reviewer: ```suggestion
- Avoid requiring the fairscale package to use precision with the fsdp native strategy ([#14092](https://github.com/Lightning-AI/lightning/pull/14092))
```

- Fixed saving hyperparameters in a composition where the parent class is not a `LightningModule` or `LightningDataModule` ([#14151](https://github.com/Lightning-AI/lightning/pull/14151))

> - Avoided requiring the fairscale package to use precision with the fsdp native strategy ([#14092](https://github.com/Lightning-AI/lightning/pull/14092))

Reviewer 2: ```suggestion
- Avoided requiring the FairScale package to use precision with the fsdp native strategy ([#14092](https://github.com/Lightning-AI/lightning/pull/14092))
```
